packages/amplify-cli/src/input-manager.ts

```diff
@@ -63,6 +64,40 @@ export function getCommandLineInput(pluginPlatform: PluginPlatform): Input {
   return result;
 }
 
+function preserveHelpInformation(input: Input): Input {
+  const subCommands = input.subCommands ? input.subCommands : [];
+  // preserve non-help command in subcommands
+  if (input.command && input.command.toLowerCase() !== constants.HELP) {
+    subCommands.unshift(input.command.toLocaleLowerCase());
+  }
+
+  const hasLongHelpOption = typeof input.options?.[constants.HELP] === 'string';
+  const hasShortHelpOption = typeof input.options?.[constants.HELP_SHORT] === 'string';
+  // prevent information in help option from being overwritten to true by saving it in subcommands
+  if (hasLongHelpOption) {
+    subCommands.push(input.options?.[constants.HELP] as string);
+  } else if (hasShortHelpOption) {
+    subCommands.push(input.options?.[constants.HELP_SHORT] as string);
```

Comment:
nit: extraneous optional chain operators (values are checked on lines 74-75)

```suggestion
    subCommands.push(input.options.[constants.HELP] as string);
  } else if (hasShortHelpOption) {
    subCommands.push(input.options.[constants.HELP_SHORT] as string);
```